### PR TITLE
fix(datepicker): fix timezone issue

### LINKED ISSUE #3609: DatePicker timezone problem
`[bsConfig]="{ dateInputFormat: 'YYYY-MM-DD' }"`

when this format is used, i may expect the same entry come back regardless of client. because we are selecting a date not datetime. But its considering it as datetime and puts timezone info to it.

So, for example, 2018-01-30 00:00:00 becomes 2018-01-29 if your timezone is +

There is no config option to set timezone or any thing that allows to retreive the input directly. 

If there is a way, i am not aware but per the source code, i dont see any option that lets this.

### PR DESCRIPTION
**Pull request by Domainv**
Closes #3609

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [ ] added/updated tests.
 - [ ] added/updated API documentation.
 - [ ] added/updated demos.

### CODE REVIEW

File: src/datepicker/bs-datepicker-inline.component.ts [hunk 1]
```diff
@@ -23,8 +23,16 @@ export class BsDatepickerInlineDirective implements OnInit, OnDestroy, OnChanges
     if (this._bsValue === value) {
       return;
     }
-    this._bsValue = value;
-    this.bsValueChange.emit(value);
+
+    let date = value;
+
+    if (date) {
+      const timezoneOffset = value.getTimezoneOffset() * 60000;
```
Comment (imben1109) [inline src/datepicker/bs-datepicker-inline.component.ts]: Is it possible to have service to configure custom timezone offset?

File: src/datepicker/bs-datepicker.component.ts [hunk 2]
```diff
@@ -69,8 +69,15 @@ export class BsDatepickerDirective implements OnInit, OnDestroy, OnChanges {
     if (this._bsValue === value) {
       return;
     }
-    this._bsValue = value;
-    this.bsValueChange.emit(value);
+    let date = value;
+
+    if (date) {
+      const timezoneOffset = value.getTimezoneOffset() * 60000;
```
Comment (imben1109) [inline src/datepicker/bs-datepicker.component.ts]: Is this possible to have a service to have a custom timezone offset?

**Review by imben1109 — COMMENTED** [review]
Yes. It fix the UTC timezone offset issue. it would reset timezone offset to 0. 
But the could be possible situation that is would be a custom timezone offset. 
For example, date range picker in Hong Kong (UTC +8).

I would like to select a date from 27/08/2018 - 28/08/2018
the UTC Date Object should be 26/08/2018 16:00- 27/08/2018 16:00

**Comment by ludmilanesvitiy** [comment]
Peek 2019-08-27 11-04
When the user manually set date, it converts to the previous day.

**Comment by ludmilanesvitiy** [comment]
Same shift with min-mode
Screenshot from 2019-08-27 11-14-33

**Comment by ludmilanesvitiy** [comment]
Smoke tested, looks good.

File: src/datepicker/bs-datepicker-inline.component.ts (hunk 1 above)
Comment (Domainv) [reply src/datepicker/bs-datepicker-inline.component.ts]: Maybe as different feature-request

File: src/datepicker/bs-datepicker.component.ts (hunk 2 above)
Comment (Domainv) [reply src/datepicker/bs-datepicker.component.ts]: Maybe as different feature-request

**Pull request merged**